src/main/java/com/chikli/hudson/plugin/naginator/NaginatorListener.java

```diff
@@ -58,16 +58,45 @@ public void onCompleted(AbstractBuild<?, ?> build, TaskListener listener) {
             if ((regexpForRerun !=null) && (!regexpForRerun.equals(""))) {
                 LOGGER.log(Level.FINEST, "regexpForRerun - {0}", regexpForRerun);
 
-                try {
-                    // If parseLog returns false, we didn't find the regular expression,
-                    // so return true.
-                    if (!parseLog(build.getLogFile(), regexpForRerun)) {
-                        LOGGER.log(Level.FINEST, "regexp not in logfile");
-                        return;
-                    }
-                } catch (IOException e) {
-                    e.printStackTrace(listener
+                if (build instanceof MatrixBuild) {
+                    MatrixBuild mb = (MatrixBuild)build;
+                    List<MatrixRun> matrixRuns = mb.getRuns();
+
+					// for matrix builds, check the logs of all matrix runs
```

Comment:
Inconsist indentation.
Please make your editor display spacing characters and see letters for indentation
(unfortunately, github doesn't display them).